Home_Work_6/chess/check_queen.py

```diff
@@ -0,0 +1,41 @@
+# Добавьте в пакет, созданный на семинаре шахматный модуль.
+# Внутри него напишите код, решающий задачу о 8 ферзях.
+# Известно, что на доске 8×8 можно расставить 8 ферзей так, чтобы они
+# не били друг друга. Вам дана расстановка 8 ферзей на доске, определите,
+# есть ли среди них пара бьющих друг друга. Программа получает на вход восемь
+# пар чисел, каждое число от 1 до 8 - координаты 8 ферзей. Если ферзи не бьют
+# друг друга верните истину, а если бьют - ложь.
+# Напишите функцию в шахматный модуль. Используйте генератор случайных чисел
+# для случайной расстановки ферзей в задаче выше. Проверяйте различные случайные
+# варианты и выведите 4 успешных расстановки. *Выведите все успешные варианты
+# расстановок
+
+from itertools import permutations as perm, combinations_with_replacement as comb, combinations
+
+
+def check_queen(coordinates: list) -> bool:
+    for item in coordinates:
+        for j in range(coordinates.index(item) + 1, len(coordinates)):
+            # проверка диагоналей, затем горизонтали и вертикали
+            if item[0] - item[1] == coordinates[j][0] - coordinates[j][1] or \
+                    item[0] + item[1] == coordinates[j][0] + coordinates[j][1] or \
+                    item[0] == coordinates[j][0] or item[1] == coordinates[j][1]:
+                return False
+    return True
+
+
+def generator_and_print():
+    variants = sorted(list(set(list(comb(range(1, 9), 2)) + list(perm(range(1, 9), 2)))))
```

Comment:
Очень большой массив с координатами. Реальных комбинаций положения ферзей будет значительно меньше. Их всего 64, а уже потом эти 64 положения по 8 штук может быть несколько миллионов. Их вы получаете в 30 строке. А тут можно вот так
```suggestion
    variants = (x, y for x in range(1, 9) for y in range(1, 9))
```